hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1882,26 +1905,26 @@ int QuectelNcpClient::modemPowerOn() {
     if (!modemPowerState()) {
         ncpPowerState(NcpPowerState::TRANSIENT_ON);
 
-        if (ncpId() != PLATFORM_NCP_QUECTEL_BG95_M1 && ncpId() != PLATFORM_NCP_QUECTEL_BG95_MF) {
-            // Power on, power on pulse >= 100ms
-            // NOTE: The BGPWR pin is inverted
-            hal_gpio_write(BGPWR, 1);
-            HAL_Delay_Milliseconds(200);
-            hal_gpio_write(BGPWR, 0);
-        } else {
-            // Power on, power on pulse 500ms - 1000ms
-            // XXX: Keeping it halfway between 500ms and 650ms, to avoid the power OFF timing of >=650ms
-            // NOTE: The BGPWR pin is inverted
-            hal_gpio_write(BGPWR, 1);
-            HAL_Delay_Milliseconds(575);
-            hal_gpio_write(BGPWR, 0);
-        }
+        // EG91, BG96, BG95, BG77: Power on, power on pulse 500ms - 1000ms
+        // XXX: Keeping it halfway between 500ms and 650ms, to avoid the power OFF timing of >=650ms
+        // NOTE: The BGPWR pin is inverted
+        hal_gpio_write(BGPWR, 1);
+        HAL_Delay_Milliseconds(575);
+        hal_gpio_write(BGPWR, 0);
 
         // After power on the device, we can't assume the device is ready for operation:
-        // BG95: status pin ready requires >= 2.1s, uart ready requires >= 2.5s
+        // BG95/BG77: status pin ready requires >= 2.1s, uart ready requires >= 2.5s
         // BG96: status pin ready requires >= 4.8s, uart ready requires >= 4.9s
         // EG91: status pin ready requires >= 10s, uart ready requires >= 12s
-        if (waitModemPowerState(1, 15000)) {
+        // Wait 3 seconds longer than minimum spec
+        int ncp_id = ncpId();
```

Comment:
Nit: It's fine to just `ncpId() ==` without temp variable, it'll be optimized just fine.